research-software-directory/RSD-as-a-service
# Release scraper and software citation section

# Release scraper and software citation section

Closes #711
Closes #713 

Changes proposed in this pull request:
* Release scraper is changed to use DOI to extract information about releases. The version is extracted from DOI. In case version is not available scraper tries to extract it from github repo branch registered in DOI. The citation formats are not scraped anymore. 
* Release reporting on organisation page is updated to use information from new scraper
* Citation section on software page is updated to use [doi.org content negotiation](https://citation.crosscite.org/docs.html) to generate citation formats for download. This change resulted in adapted offer of citation formats. 
* Unit test are updated 

Known issues:
- The version is not provided in all DOI. Based on test data mainly DOI created before 2019 might not have version value (null). Improvement suggestions are welcome

How to test:
* `make start` to rebuild everything
* `docker-compose down && docker-compose up`: restart to enable release scraper
* wait until release scraper does it's job
* verify release report on organisation page
* verify releases list on software page (all versions present?)
* verify citation exports are working

## Example release report per organisation
![image](https://user-images.githubusercontent.com/9204081/216823596-be879a04-4a3e-4dcd-adb1-2619c2bd74c5.png)

## Example software citation section
![image](https://user-images.githubusercontent.com/9204081/216826691-b857503f-1b29-4345-af3a-97b114c0d639.png)

PR Checklist:
